torch/_inductor/codegen/cpp.py

```diff
@@ -1209,8 +1209,19 @@ def scale_index_with_offset(
         new_index = sympy_subs(index, replacement)
         return new_index
 
-    @staticmethod
-    def indirect_indexing(index_var, size, check=True):
+    def indirect_indexing(self, index_var, size, check=True):
+        generate_assert = (
+            (check or config.debug_index_asserts)
+            and config.cpp.assert_indirect_indexing
+        )
+
+        if generate_assert:
+            expr = ops.and_(ops.ge(index_var, '0'),
+                            ops.le(index_var, self.rename_indexing(size)))
+            cond = self.cse.generate(self.compute, expr)
+            line = f'assert({cond} && "index out of bounds");'
+            self.compute.writeline(line)
```

Comment:
Could we also share the logic where, if we find a tighter bound on a variable later on, we elide the check?